modules/common/src/main/java/org/opencastproject/mediapackage/MediaPackageElementFlavor.java

```diff
@@ -392,52 +254,27 @@ public MediaPackageElementFlavor unmarshal(String str) throws Exception {
   }
 
   public boolean matches(MediaPackageElementFlavor other) {
-    if (other == null)
-      return false;
-    if (this == other)
-      return true;
-    if (subtype == null) {
-      if (other.subtype != null && !isWildcard(other.subtype))
-        return false;
-    } else if (!subtype.equals(other.subtype) && (!isWildcard(subtype) && !isWildcard(other.subtype)))
-      return false;
-    if (type == null) {
-      if (other.type != null && !isWildcard(other.type))
-        return false;
-    } else if (!type.equals(other.type) && (!isWildcard(type) && !isWildcard(other.type)))
-      return false;
-    return true;
+    return (other != null) && matchParts(type, other.type) && matchParts(subtype, other.subtype);
+  }
+
+  private boolean matchParts(String part1, String part2) {
+    return part1.equals(part2) || isWildcard(part1) || isWildcard(part2);
   }
 
   @Override
   public int hashCode() {
     final int prime = 31;
     int result = 1;
-    result = prime * result + ((subtype == null) ? 0 : subtype.hashCode());
-    result = prime * result + ((type == null) ? 0 : type.hashCode());
+    result = prime * result + subtype.hashCode();
```

Comment:
We don't even need to initialize this before:

```suggestion
    int result = prime + subtype.hashCode();
```

Reply:
Thinking about it again, the whole method is just:

```java
public int hashCode() {
￼     return 961 + subtype.hashCode() + type.hashCode();
}
```

Reply:
But this doesn't explain anymore that a prime is used to build the hash code... the code should be easy to understand - not necessarily as short as possible.

Reply:
True but irrelevant because it does not really matter that it's a prime number, does it?
In both cases, it's basically only a randomly chosen magic number.

But if you want to keep it, maybe:

```java
final int prime = 31;
return prime * prime + subtype.hashCode() + type.hashCode();
```

Because the `result` variable does not make it easier to read.

Reply:
Shortly looked into this. With Java7+, this should be what we really want to use instead of writing our own code (sorry for all the updates):

```java
return Objects.hash(subtype, type);
```

Reply:
I did some quick research but it does not seem 100% clear whether Objects.hash is a good choice.

Maybe it is, but I don't really have time to look into this. While this code could be further simplified, it is not as terrible as some other code here was...